### ISSUE TITLE
[block-spacing]: new option 'consistent'

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13175]
**What rule do you want to change?**
`block-spacing`

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new options named `consistent`
Currently, as this is a layout type rule, I think it is good to have a consistent space is first and last curly braces for block statements.

**Please provide some example code that this change will affect:**
```js
/*eslint block-spacing: "error"*/

function foo() { return true; } // lint free
if (foo) {       bar = 0; }  // error as there is inconsistent between number of spaces in before and after the body
```
this option wont be applicable for `never` option

```js
/*eslint block-spacing: "never"*/

function foo() {return true;} // lint free
if (foo) {bar = 0; }  // usual error no after error. 
```

**What does the rule currently do for this code?**
it is lint-free currently

**What will the rule do after it's changed?**
it will report whenever its `always` as the first option and there is inconsistency in the number of spaces. It will ignore a newline like this causes this rule mainly focuses on spaces rather than lines. 

```js
/*eslint block-spacing: "error"*/
function foo() { // lint free
return; } // lint free
```

**Are you willing to submit a pull request to implement this change?**
Yes,

**Comment by mdjermanovic** [comment]
@anikethsaha thanks for the issue!

For a single space, this should be already covered by the [no-multi-spaces](https://eslint.org/docs/rules/no-multi-spaces) rule:

```js
/*eslint block-spacing: "error"*/
/*eslint no-multi-spaces: "error"*/

if (foo) {       bar = 0; } // no-multi-spaces error
```

will be auto-fixed to:

```js
if (foo) { bar = 0; }
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgYmxvY2stc3BhY2luZzogXCJlcnJvclwiKi9cbi8qZXNsaW50IG5vLW11bHRpLXNwYWNlczogXCJlcnJvclwiKi9cblxuaWYgKGZvbykgeyAgICAgICBiYXIgPSAwOyB9IC8vIG5vLW11bHRpLXNwYWNlcyBlcnJvciIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiaW5kZW50IjoyfSwiZW52Ijp7fX19)

We don't have any rules/options to enforce multiple spaces as it doesn't seem to be a common style (unless there's some alignment like in the `key-spacing` rule, but that isn't the case with this rule).

**Comment by anikethsaha** [comment]
thanks for the input. I do think even the new line consistent is solved using `padded-block`.
